File: CH9/StartWithJEE8/StartWithJEE8-ejb/src/main/java/com/daniel/delivery/abstraction/service/delivery/DeliveryService.java

package com.daniel.delivery.abstraction.service.delivery;

import com.daniel.delivery.abstraction.entity.Delivery;
import java.util.List;

public interface DeliveryService {
    Delivery request(Delivery delivery);
    List<Delivery> getByEmailAndState(String email, String state);
    
}

File: CH10/startwithvuejs/src/services/DeliveryService.ts
import Vue from 'vue'
import { Delivery } from '../entities/Delivery'

export class DeliveryService {
  static getDeliveriesByEmailAndState (email:string, state:string) {
    return Vue.axios.get<Array<Delivery>>(`/deliveries/${email}?state=${state}`)
  }

  static request (delivery: Delivery) {
    return Vue.axios.post<Delivery>(`/deliveries`, delivery)
  }
}
